sites/kit.svelte.dev/src/lib/search/SearchBox.svelte

```diff
@@ -0,0 +1,416 @@
+		const suffix = content.slice(
+			index + query.length,
+			index + query.length + (80 - (prefix.length + query.length))
+		);
+
+		return (
+			escape(prefix) +
+			`<mark>${escape(content.slice(index, index + query.length))}</mark>` +
+			escape(suffix)
+		);
+	}
+
+	/** @param {string} href */
+	function navigate(href) {
+		$recent = [href, ...$recent.filter((x) => x !== href)];
+		$searching = false;
+	}
+
+	$: if ($searching) update();
+
+	$: recent_searches = lookup ? $recent.map((href) => lookup.get(href)).filter(Boolean) : [];
+</script>
+
+<svelte:window
+	on:keydown={(e) => {
+		if (e.code === 'KeyK' && (navigator.platform === 'MacIntel' ? e.metaKey : e.ctrlKey)) {
+			e.preventDefault();
+			$query = '';
+			$searching = !$searching;
+		}
+
+		if (e.code === 'Escape') {
+			$searching = false;
+		}
+	}}
+/>
+
+{#if $searching && index}
+	<div
+		class="modal-background"
+		on:click={() => ($searching = false)}
+		on:wheel={(e) => e.preventDefault()}
+		on:touchmove={(e) => e.preventDefault()}
+	/>
+
+	<div
+		bind:this={modal}
+		class="modal"
+		on:keydown={(e) => {
+			if (e.key === 'ArrowDown' || e.key === 'ArrowUp') {
+				e.preventDefault();
+				const group = focusable_children(e.currentTarget);
+
+				// when using arrow keys (as opposed to tab), don't focus buttons
+				const selector = 'a, input';
+
+				if (e.key === 'ArrowDown') {
+					group.next(selector);
+				} else {
+					group.prev(selector);
+				}
+			}
+
+			if (e.code === 'Space' && e.target.nodeName !== 'INPUT') {
+				e.preventDefault();
+			}
+		}}
+		use:trap
+	>
+		<div class="search-box">
+			<!-- svelte-ignore a11y-autofocus -->
+			<input
+				autofocus
+				on:keydown={(e) => {
+					if (e.key === 'Enter') {
+						if (results.length > 0) {
+							modal.querySelector('a').click();
+						}
+					}
+
+					// ideally, opening the modal would create a history entry that we
+					// could use with browser back/forward buttons — this would be the
+					// best way to allow people to close the modal on mobile, for
+					// example. That's a little tricky to do right now — need to add
+					// new APIs to SvelteKit — but in the meantime we can close the
+					// modal when the backspace key is pressed. we _don't_ want that
+					// to happen when the backspace keydown fires repeatedly, so we
+					// track whether it was already pressed or not. (Sadly this
+					// doesn't work on mobile, longpress will close the modal)
+					if (e.key === 'Backspace') {
+						if ($query === '' && !backspace_pressed) {
+							$searching = false;
+						}
+
+						backspace_pressed = true;
+					}
+				}}
+				on:keyup={(e) => {
+					if (e.key === 'Backspace') {
+						backspace_pressed = false;
+					}
+				}}
+				on:input={(e) => {
+					$query = e.target.value;
+					update();
+				}}
+				value={$query}
+				placeholder="Search"
+			/>
+
+			<button aria-label="Close" on:click={() => ($searching = false)}>
```

Comment:
This is misaligned on iOS Safari:
![close](https://user-images.githubusercontent.com/4992896/152662115-0bb6dd1f-7879-474f-a46e-020f0ecaa0b1.jpg)

Reply:
Ugh safari. It was broken on desktop safari too, which i fixed by adding `position: relative` to `.search-box`. Hopefully that fixes iOS safari as well (will be able to check later when i can borrow an iphone)

Reply:
This is fixed now, though there's on more Safari issue I noticed -- seems like there's an overlapping icon on the search box. ![image](https://user-images.githubusercontent.com/4992896/153256380-1a55d405-90aa-4098-b2d4-8f82fa126bc0.jpeg)